PayloadPkg/OsLoader/OsLoader.c

```diff
@@ -497,10 +497,15 @@ StartBooting (
       EntryPoint = (PRE_OS_CHECKER_ENTRY) (UINTN)mPreOsCheckerEntry;
       BeforeOSJump ("Starting Pre-OS Checker ...");
 
-      OsBootState.Esi = (UINT32) BootParams;
-      OsBootState.Eip = BootParams->Hdr.Code32Start;
+      PreOsParams.Version  = 0x1;
+      PreOsParams.HeapSize = EFI_SIZE_TO_PAGES (0);
+      PreOsParams.HeapAddr = (UINT32) AllocatePages (PreOsParams.HeapSize);
```

Comment:
Just want to confirm that the memory space for preOs heap could be used by normal OS later, right? 
If not, we need update normal OS memory map.

Reply:
That should be correct, all of the memory used by the pre-OS payload/checker can be reclaimed/used by the OS, it should not require any reserved memory allocation/map-entries.